realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -346,14 +313,56 @@ protected void onError(LogoutResponse response) {
     }
 
     /**
+     * Change this users password. This is done synchronously and involves the network, so calling this method on the
+     * Android UI thread will always crash.
+     *
+     * @param newPassword the users new password.
+     * @throws ObjectServerError if the password could not be changed for some reason.
+     */
+    public void changePassword(String newPassword) throws ObjectServerError {
+        if (newPassword == null) {
```

Comment:
Yes, I didn't want to encode any requirements on the client side. The server should return an error if the password isn't good enough. It is arguable if we should even throw on `null`